targets/TARGET_NUVOTON/TARGET_M480/trng_api.c

```diff
@@ -77,19 +82,20 @@ void trng_free(trng_t *obj)
 int trng_get_bytes(trng_t *obj, uint8_t *output, size_t length, size_t *output_length)
 {
     (void)obj;
-
+    unsigned char tmpBuff[32];
+    
     *output_length = 0;
-    if (length < 32) {
-        unsigned char tmpBuff[32];
+    
+    for (unsigned i = 0; i < (length/32); i++) {
+        trng_get(output);
+        *output_length += 32;
+        output += 32;
+    }     
+    if( length > *output_length ) {
+        trng_zeroize(tmpBuff, sizeof(tmpBuff));
```

Comment:
@cyliangtw Thanks for the quick reworking. The zeroization should be done after the bytes have been fetched from the TRNG.

Reply:
@hanno-arm , after fetch from TRNG and copy to output, tmpBuffer is no longer used in this function. Why still need pay CPU resource to do zeroization, do you concern to disclose partial of random number in stack memory ?

Reply:
@cyliangtw Yes, exactly. The zeroization at the end (which is the only one that's needed here) is to prevent leakage of the random data on the stack - imagine the OS handing the same stack memory to some other process later. That's why one generally wipes sensitive data from the stack before returning.